### PR TITLE
feat(lombok): making lombok.jar available to java sidecar

### PR DESCRIPTION
**Pull request by sunix**
### What does this PR do?
Making the lombok.jar available to Java.
So user would just need to configure in the devfile:
```
  - id: redhat/java/latest
    type: chePlugin
    preferences:
      java.jdt.ls.vmargs: -javaagent:/home/theia/lombok.jar
```

### Screenshot/screencast of this PR
Selection_249

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17483

### How to test this PR?
Use this devfile: it is using a custom registry that has the java plugin patched to use the image of the sidecar i have built locally.
1. start a workspace with the following devfile
2. open one of the java file `ModelLombokValue.java`, wait for java plugin to be activated
3. open the outline, see the getters and setters that are not defined in the class.

```yaml
apiVersion: 1.0.0
metadata:
  generateName: java-lombok-
projects:
  - name: lombok-project-example
    source:
      location: 'https://github.com/chuucks/LOMBOK-PROJECT-EXAMPLE'
      type: git
      branch: master

components:

  - type: chePlugin
    id: redhat/java/latest
    preferences:
      java.server.launchMode: Standard
      java.jdt.ls.vmargs: -javaagent:/home/theia/lombok.jar
    registryUrl: https://sunix-plugin-registry-java-lombok-agher.surge.sh/v3

  - mountSources: true
    endpoints:
      - name: debug
        port: 5005
        attributes:
          public: 'false'
    memoryLimit: 512Mi
    type: dockerimage
    volumes:
      - name: m2
        containerPath: /home/user/.m2
    image: 'quay.io/eclipse/che-java11-maven:7.22.1'
    alias: maven
    env:
      - value: ''
        name: MAVEN_CONFIG
      - value: >-
          -XX:MaxRAMPercentage=50 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
          -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4
          -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true
          -Xms20m -Djava.security.egd=file:/dev/./urandom -Duser.home=/home/user
        name: MAVEN_OPTS
      - value: >-
          -XX:MaxRAMPercentage=50 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
          -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4
          -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true
          -Xms20m -Djava.security.egd=file:/dev/./urandom
        name: JAVA_OPTS
      - value: >-
          -XX:MaxRAMPercentage=50 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
          -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4
          -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true
          -Xms20m -Djava.security.egd=file:/dev/./urandom
        name: JAVA_TOOL_OPTIONS

commands:
  - name: maven build
    actions:
      - workdir: '${CHE_PROJECTS_ROOT}/lombok-project-example'
        type: exec
        command: mvn clean install
        component: maven
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: sidecars/java/Dockerfile [hunk 1]
```diff
@@ -24,6 +24,8 @@ RUN apk --no-cache add openjdk11 --repository=http://dl-cdn.alpinelinux.org/alpi
 ENV JAVA_HOME /usr/lib/jvm/default-jvm/
 ADD etc/before-start.sh /before-start.sh
 
+RUN cd ${HOME} && wget https://projectlombok.org/downloads/lombok.jar
```
Comment (benoitf) [inline sidecars/java/Dockerfile:27]: Hi, I suppose we should have a versioned file ?

File: sidecars/java/Dockerfile [hunk 2]
```diff
@@ -24,6 +24,8 @@ RUN apk --no-cache add openjdk11 --repository=http://dl-cdn.alpinelinux.org/alpi
 ENV JAVA_HOME /usr/lib/jvm/default-jvm/
 ADD etc/before-start.sh /before-start.sh
 
+RUN wget -O ${HOME}/lombok.jar https://projectlombok.org/downloads/lombok-1.18.18.jar
```
Comment (ericwill) [inline sidecars/java/Dockerfile:27]: Let's make this an `ENV` variable at the top of the file so it's easily seen. Something like `LOMBOK_VERSION` or something, here's an example: https://github.com/eclipse/che-plugin-registry/blob/master/sidecars/kubernetes-tooling/Dockerfile#L13

File: sidecars/java8/Dockerfile [hunk 3]
```diff
@@ -25,6 +25,8 @@ ADD etc/before-start.sh /before-start.sh
 
 ADD etc/entrypoint.sh /entrypoint.sh
 
+RUN wget -O ${HOME}/lombok.jar https://projectlombok.org/downloads/lombok-1.18.18.jar
```
Comment (ericwill) [inline sidecars/java8/Dockerfile:28]: Same comment about the version as above.

File: sidecars/java/Dockerfile (hunk 2 above)
Comment (sunix) [reply sidecars/java/Dockerfile]: I updated the PR

**Review by ericwill — APPROVED** [review]
Works as expected on che.openshift.io. :tada: 

One small comment: there is a popup when I first launch the workspace, asking to "allow" the jdt.ls setting. Is this also expected?

**Comment by sunix** [comment]
> One small comment: there is a popup when I first launch the workspace, asking to "allow" the jdt.ls setting. Is this also expected?

I had that popup as well, maybe something to ask to the vscode-java team.

**Comment by benoitf** [comment]
I've rebased the PR of sun with my fix on skipping digest check when we modify sidecars.

We can check images for java and java8 are the one for the upcoming sha-1 and that it passed build
https://pr-check-860-alpine-che-plugin-registry.surge.sh/v3/external_images.txt

**Review by benoitf — APPROVED** [review]
it would have been better with different sidecars/entrypoint/mount/etc but it may not be possible to implement easily.

**Pull request merged**